validator_client/src/config.rs

```diff
@@ -1,32 +1,71 @@
+use bls::Keypair;
 use std::fs;
+use std::fs::File;
+use std::io::{Error, ErrorKind};
 use std::path::PathBuf;
 use types::ChainSpec;
 
 /// Stores the core configuration for this validator instance.
 #[derive(Clone)]
 pub struct ClientConfig {
     pub data_dir: PathBuf,
+    pub key_dir: PathBuf,
     pub server: String,
     pub spec: ChainSpec,
 }
 
 const DEFAULT_LIGHTHOUSE_DIR: &str = ".lighthouse-validators";
+const DEFAULT_KEYSTORE_SUBDIR: &str = "keystore";
 
 impl ClientConfig {
     /// Build a new configuration from defaults.
-    pub fn default() -> Self {
+    pub fn default() -> Result<Self, Error> {
         let data_dir = {
             let home = dirs::home_dir().expect("Unable to determine home dir.");
```

Comment:
It's pretty minor, but maybe we should get rid of this `expect(..)` and map it back to a `std::io:Error` so it's consistent with the `create_dir_all` calls.

I'm thinking it could be an `ErrorKind::NotFound`?